fsspec/implementations/memory.py

```diff
@@ -25,6 +26,9 @@ class MemoryFileSystem(AbstractFileSystem):
 
     @classmethod
     def _strip_protocol(cls, path):
+        path = stringify_path(path)
+        if "\\" in path:
+            path = path.replace("\\", "/")
```

Comment:
I had meant to ask: why is this necessary?

Reply:
I was having an issue with the way paths work on Windows. When Path is converted to a string on Windows, it will use "\". I ended up with a really weird path that looked like "/\\myfile.pdf". So that step makes it "//myfile.pdf", and then some code later on in the function strips out the duplicate slash.

Reply:
Yes, MemoryFileSystem assumes posix. I think we should still allow Path, because the base `AbstractFileSystem` class stringifies paths. It would be nice to have `MemoryFileSystem` be consistent in that regard. Should I move the posixifyl logic to utils, or just directly import from `LocalFileSystem`?

Reply:
Actually, there's no problem with the "weird" path in memFS at all. Why not just leave it? If you supply a Path, you get OS-specific characters. That doesn't seem so bad. 

-edit: please ignore below here-

Hm, but `Path` says it only applies to local filesystems.

> When Path is converted to a string on Windows, it will use "\".

When is this conversion happening? Perhaps we could replace this and the line above
```suggestion
            if isinstance(path, Path):
                path = LocalFileSystem._strip_protocol(path)
```
(so this only happens in the rare case)